src/games/crazy_eights.rs

```diff
@@ -0,0 +1,632 @@
+use crate::common::deck::STANDARD_DECK;
+use crate::common::rand::RngSeed;
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Player(pub u8);
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, Serialize_repr, Deserialize_repr)]
+#[repr(u8)]
+pub enum NumberOfPlayers {
+    Two = 2,
+    Three = 3,
+    Four = 4,
+}
+
+impl NumberOfPlayers {
+    /// Returns the number of players for the current game type
+    /// ```
+    /// use lib_table_top::games::crazy_eights::NumberOfPlayers;
+    ///
+    /// assert_eq!(NumberOfPlayers::Two.to_int(), 2);
+    /// assert_eq!(NumberOfPlayers::Three.to_int(), 3);
+    /// assert_eq!(NumberOfPlayers::Four.to_int(), 4);
+    /// ```
+    pub fn to_int(&self) -> u8 {
+        match self {
+            NumberOfPlayers::Two => 2,
+            NumberOfPlayers::Three => 3,
+            NumberOfPlayers::Four => 4,
+        }
+    }
+
+    /// Returns the starting number of cards per player
+    /// ```
+    /// use lib_table_top::games::crazy_eights::NumberOfPlayers::*;
+    ///
+    /// assert_eq!(Two.starting_number_of_cards_per_player(), 7);
+    /// assert_eq!(Three.starting_number_of_cards_per_player(), 5);
+    /// assert_eq!(Four.starting_number_of_cards_per_player(), 5);
+    /// ```
+    pub fn starting_number_of_cards_per_player(&self) -> u8 {
+        match self {
+            NumberOfPlayers::Two => 7,
+            NumberOfPlayers::Three => 5,
+            NumberOfPlayers::Four => 5,
+        }
+    }
+
+    /// An iterator of players for a game type. (Players are 0 indexed)
+    /// ```
+    /// use lib_table_top::games::crazy_eights::{NumberOfPlayers, Player};
+    ///
+    /// assert_eq!(
+    ///   NumberOfPlayers::Two.players().collect::<Vec<Player>>(),
+    ///   vec![Player(0), Player(1)]
+    /// );
+    ///
+    /// assert_eq!(
+    ///   NumberOfPlayers::Four.players().collect::<Vec<Player>>(),
+    ///   vec![Player(0), Player(1), Player(2), Player(3)]
+    /// );
+    /// ```
+    pub fn players(&self) -> impl Iterator<Item = Player> + Clone {
+        (0..self.to_int()).map(|num| Player(num))
+    }
+}
+
+#[derive(Clone, Copy, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Settings {
+    pub seed: RngSeed,
+    pub number_of_players: NumberOfPlayers,
+}
+
+#[derive(Clone, Debug, Hash, PartialEq, Eq, Serialize, Deserialize)]
+pub struct GameHistory {
+    settings: Settings,
+    history: Vec<Action>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct GameState {
+    game_history: GameHistory,
+    rng: ChaCha20Rng,
+    discarded: Vec<Card>,
+    hands: HashMap<Player, Vec<Card>>,
+    draw_pile: Vec<Card>,
+    top_card: Card,
+    current_suit: Suit,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Status {
+    InProgress,
+    Win { player: Player },
+}
+
+use Status::*;
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct PlayerView<Container>
```

Comment:
I'd rather this be a container like `Vec` instead of a concrete `Vec<Card>`, but it seems like it's not supported yet

https://github.com/rust-lang/rust/issues/44265